pkg/operator/collection.go

```diff
@@ -263,6 +263,9 @@ func (r *collectionReconciler) makeCollectorDaemonSet(spec *monitoringv1alpha1.C
 	for _, matcher := range spec.Filter.MatchOneOf {
 		collectorArgs = append(collectorArgs, fmt.Sprintf("--export.match=%s", matcher))
 	}
+	if r.opts.OperatorDeployment != "" {
+		collectorArgs = append(collectorArgs, fmt.Sprintf("--export.user-agent=prometheus-collector/%s (deployment:%s)", Version, r.opts.OperatorDeployment))
```

Comment:
We want this `Version` to be the Prometheus version (e.g. [`2.28.1-gmp.7`](https://github.com/GoogleCloudPlatform/prometheus/blob/release-2.28.1-gmp/cmd/prometheus/main.go#L325) in our prometheus fork) rather than the `prometheus-engine-export` version (which will already [be prepended](https://github.com/GoogleCloudPlatform/prometheus-engine/blob/492b943b5edbf467798cbf616ceb408d3fc1d0c8/pkg/export/export.go#L216-L217) in the full user agent).